### PR TITLE
Refactoring for better testing

### PR DESCRIPTION
**Pull request by nrc**
### What problem does this PR solve? 

This PR is a step towards making unit tests more unit-y and a better long-term test framework

cc #111

PTAL @lonng 

### What is changed and how it works?

The first commit refactors the list component so that it can be unit tested without also testing the generic running framework or the table writing code.

The second commit refactors the `profile` and `repo` globals into an `Environment` struct. This removes global, mutable state and will allow for tests to run concurrently using different environments.

### Check List 

Tests 

 - Unit test
 - Integration test
 
(existing tests, all work is refactoring)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

none

Related changes

none

### CODE REVIEW

**Comment by nrc** [comment]
If I understand the test output, the failure is due to code coverage shrinking, but this is intentional because the unit tests are testing less. `make test` passes locally

**Comment by lonng** [comment]
@nrc 
image

File: components/playground/instance/instance.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package instance
 import (
 	"context"
 	"fmt"
+	"github.com/pingcap-incubator/tiup/pkg/localdata"
```
Comment (lonng) [inline components/playground/instance/instance.go:19]: Please format the imports.

File: pkg/meta/meta.go [hunk 2]
```diff
@@ -62,110 +34,63 @@ func Mirror() string {
...
-func Repository() *repository.Repository {
-	return repo
+func (env Environment) Repository() *repository.Repository {
+	return env.repo
 }
 
 // SetRepository exports for test
-func SetRepository(r *repository.Repository) {
-	repo = r
+func (env Environment) SetRepository(r *repository.Repository) {
```
Comment (lonng) [inline pkg/meta/meta.go:68]:
```suggestion
func (env *Environment) SetRepository(r *repository.Repository) {
```

File: pkg/meta/meta.go [hunk 3]
```diff
@@ -62,110 +34,63 @@ func Mirror() string {
...
-func Profile() *localdata.Profile {
-	return profile
+func (env Environment) Profile() *localdata.Profile {
+	return env.profile
 }
 
 // SetProfile exports for test
-func SetProfile(p *localdata.Profile) {
-	profile = p
+func (env Environment) SetProfile(p *localdata.Profile) {
```
Comment (lonng) [inline pkg/meta/meta.go:78]: The receiver must a pointer, or else the `set` will take no effect.
```suggestion
func (env *Environment) SetProfile(p *localdata.Profile) {
```

**Comment by lonng** [comment]
Seems there are some integration tests unstable.
image

The expectation and the got result indeed the same. It's weird.

File: components/playground/instance/instance.go (hunk 1 above)
Comment (nrc) [reply components/playground/instance/instance.go:19]: Sorry, I always forget to do this. I've added goimports to `make fmt`, I hope that helps.

**Comment by nrc** [comment]
The doc help failure is due to 00d033b0ec8b85d29ae0bb73024fc7d45781acc9, there is a fix in the new 3rd commit. I think the instability is due to using `Environment` and not `*Environment`.

File: pkg/meta/meta.go [hunk 4]
```diff
@@ -14,45 +14,17 @@
 package meta
 
 import (
+	"github.com/pingcap-incubator/tiup/pkg/localdata"
```
Comment (lonng) [inline pkg/meta/meta.go:17]: Place all `std` library imports at the front of the 3rd will be more idiomatic.

Comment (nrc) [reply pkg/meta/meta.go]: I wish goimports did this, or even better, gofmt.

**Pull request merged**